pontoon/base/models.py

```diff
@@ -2762,8 +2814,45 @@ def latest_activity(self):
     def __str__(self):
         return self.string
 
+    def save_stats_diff(self, **stats_diff):
+        entity = self.entity
+        resource = entity.resource
+        project = resource.project
+        locale = self.locale
+
+        translated_resource, _ = TranslatedResource.objects.get_or_create(
+            resource=entity.resource,
+            locale=locale,
+        )
+
+        translated_resource.adjust_stats(**stats_diff)
+        project.adjust_stats(**stats_diff)
+
+        if not project.system_project:
+            locale.adjust_stats(**stats_diff)
+
+        project_locale = utils.get_object_or_none(
+            ProjectLocale,
+            project=project,
+            locale=locale,
+        )
+
+        if project_locale:
+            project_locale.adjust_stats(**stats_diff)
+
     def save(self, *args, **kwargs):
+        entity_states_before = self.entity.stats_states_map(self.locale)
```

Comment:
I would change names of these variables and methods to:
`stats_before = self.entity.get_stats(self.locale)`